h2o-automl/src/test/java/ai/h2o/automl/targetencoding/TargetEncodingDistributedTest.java

```diff
@@ -0,0 +1,107 @@
+package ai.h2o.automl.targetencoding;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.MRTask;
+import water.TestUtil;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.TestFrameBuilder;
+import water.fvec.Vec;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class TargetEncodingDistributedTest extends TestUtil {
+
+
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(2);
+  }
+
+  private Frame fr = null;
+
+
+  @Test
+  public void imputeNAsForColumnTest() {
+    String teColumnName = "ColA";
+    String targetColumnName = "ColB";
+    fr = new TestFrameBuilder()
+            .withName("testFrame")
+            .withColNames(teColumnName, targetColumnName)
+            .withVecTypes(Vec.T_CAT, Vec.T_CAT)
+            .withDataForCol(0, ar("a", "b", null, null, null))
+            .withDataForCol(1, ar("2", "6", "6", "2", "6"))
+            .withChunkLayout(3,2)
+            .build();
+
+    String nullStr = null;
+    fr.vec(0).set(2, nullStr);
+
+    String[] teColumns = {""};
+    TargetEncoder tec = new TargetEncoder(teColumns);
+
+    assertTrue(fr.vec("ColA").isCategorical());
+    assertEquals(2, fr.vec("ColA").cardinality());
+
+    Frame res = tec.imputeNAsForColumn(fr, "ColA", "ColA_NA");
+
+    new MRTask() {
+      @Override
+      public void map(Chunk[] cs) {
+        long levelValue = cs[0].at8(0);
+
+        System.out.println("Level value: " + levelValue);
+        System.out.println("Domain: " + Arrays.toString(cs[0].vec().domain()));
+        int length = cs[0].vec().domain().length;
+        System.out.println("Domain length: " + length);
+        cs[0].vec().factor(2); // should throw exception on one of the nodes
```

Comment:
This fails because the Vec is not properly updated in DKV in function `imputeNAsForColumn`

      ...
      System.arraycopy(oldDomain, 0, newDomain, 0, oldDomain.length);
      newDomain[indexForNACategory] = strToImpute;
      currentVec.setDomain(newDomain);
      return data;